clms-angular/src/app/core/model/User.model.ts

```diff
@@ -8,3 +8,5 @@ export type User = {
   roles: Role[];
   permissions: Permission[];
 }
+
+export type UserProjection = Partial<User>;
```

Comment:
I think this can work for now so for any request we make for a UserProjection the response can have a varying amount of information,